griptape/tasks/image_generation_task.py

```diff
@@ -59,16 +77,20 @@ def run(self) -> ImageArtifact:
             prompts=[self.input.to_text()], rulesets=self.all_rulesets, negative_rulesets=self.negative_rulesets
         )
 
-        if self.output_dir is not None:
+        if self.output_dir or self.output_file:
             self._write_to_file(image_artifact)
 
         return image_artifact
 
     def _write_to_file(self, image_artifact: ImageArtifact) -> None:
         # Save image to file. This is a temporary workaround until we update Task and Meta
         # Memory to persist artifacts from tasks.
-        outfile = path.join(self.output_dir, image_artifact.name)
+        if self.output_file:
+            outfile = self.output_file
+        else:
+            outfile = path.join(self.output_dir, image_artifact.name)
 
+        os.makedirs(self.output_dir, exist_ok=True)
```

Comment:
Should this be within the `else`? `output_dir` will be `None` if `output_file` is specified.

Reply:
Thanks! Updated to be `path.dirname(outfile)` to match the logic in tool.py